### PR TITLE
Register before start to prevent two same coordinators coexist

### PR DESCRIPTION
**Pull request by xiaocai2333**
Signed-off-by: cai.zhang <cai.zhang@zilliz.com>
issue: #21269 
/kind improvement

### CODE REVIEW

File: internal/querycoordv2/server.go [hunk 1]
```diff
@@ -156,9 +152,40 @@ func (s *Server) Init() error {
 	s.enableActiveStandBy = Params.QueryCoordCfg.EnableActiveStandby
 	s.session.SetEnableActiveStandBy(s.enableActiveStandBy)
 	Params.QueryCoordCfg.SetNodeID(s.session.ServerID)
-	Params.SetLogger(s.session.ServerID)
```
Comment (xiaofan-luan) [inline internal/querycoordv2/server.go:159]: why we remove Params.SetLogger?

Comment (xiaofan-luan) [reply internal/querycoordv2/server.go:159]: it should be in initSession?

File: internal/datacoord/server.go [hunk 2]
```diff
@@ -254,7 +254,30 @@ func (s *Server) Init() error {
...
 	s.factory.Init(&Params)
+	if err = s.initSession(); err != nil {
+		return err
+	}
+	if s.enableActiveStandBy {
+		s.activateFunc = func() {
+			log.Info("DataCoord switch from standby to active, activating")
+			if err := s.initDataCoord(); err != nil {
+				log.Warn("DataCoord init failed", zap.Error(err))
+				// TODO: panic if error occurred?
```
Comment (wayblink) [inline internal/datacoord/server.go:265]: I have another pr about active-standy. activeFunc is updated to `func() error` then we can throw this error

**Review by wayblink — APPROVED** [review]
Generally LGTM

File: internal/querycoordv2/server.go (hunk 1 above)
Comment (xiaocai2333) [reply internal/querycoordv2/server.go:159]: yes, I will make it up.

File: internal/datacoord/server.go (hunk 2 above)
Comment (xiaocai2333) [reply internal/datacoord/server.go:265]: good job, LGTM

**Pull request merged**